### ISSUE TITLE
BUG: More hours than needed are scheduled when not finding continuous hours

### ISSUE THREAD

**Issue by cnrd** [jonasbkarlsson/ev_smart_charging#82]
I just noticed that today it was scheduling 6 hours, while it only needs 1 hour to reach the targeted charge (100%) the same issues does not happen if I enable continuous hours.

I guess that the issue may be related to all of the hours chosen having the exact same price and maybe something with it crossing midnight? (I haven't looked into it much)

It may not be the highest priority bug, but I tought I would report it anyways.

![Screenshot 2022-12-30 at 18 49 45](https://user-images.githubusercontent.com/5996053/210099562-6eeb3940-5d0f-4e26-a5a7-1894ccc08c23.png)

Attributes of the sensor with continuous disabled.

```
current_price: 2.329
EV SOC: 92
EV target SOC: 100.0
Charging is planned: true
Charging start time: 2022-12-30T23:00:00+01:00
Charging stop time: 2022-12-31T07:00:00+01:00
Charging number of hours: 6
raw_two_days: 
- start: '2022-12-30T00:00:00+01:00'
  end: '2022-12-30T01:00:00+01:00'
  value: 1.584
- start: '2022-12-30T01:00:00+01:00'
  end: '2022-12-30T02:00:00+01:00'
  value: 1.536
- start: '2022-12-30T02:00:00+01:00'
  end: '2022-12-30T03:00:00+01:00'
  value: 1.517
- start: '2022-12-30T03:00:00+01:00'
  end: '2022-12-30T04:00:00+01:00'
  value: 1.494
- start: '2022-12-30T04:00:00+01:00'
  end: '2022-12-30T05:00:00+01:00'
  value: 1.494
- start: '2022-12-30T05:00:00+01:00'
  end: '2022-12-30T06:00:00+01:00'
  value: 1.532
- start: '2022-12-30T06:00:00+01:00'
  end: '2022-12-30T07:00:00+01:00'
  value: 1.591
- start: '2022-12-30T07:00:00+01:00'
  end: '2022-12-30T08:00:00+01:00'
  value: 1.713
- start: '2022-12-30T08:00:00+01:00'
  end: '2022-12-30T09:00:00+01:00'
  value: 1.746
- start: '2022-12-30T09:00:00+01:00'
  end: '2022-12-30T10:00:00+01:00'
  value: 1.759
- start: '2022-12-30T10:00:00+01:00'
  end: '2022-12-30T11:00:00+01:00'
  value: 1.731
- start: '2022-12-30T11:00:00+01:00'
  end: '2022-12-30T12:00:00+01:00'
  value: 1.718
- start: '2022-12-30T12:00:00+01:00'
  end: '2022-12-30T13:00:00+01:00'
  value: 1.713
- start: '2022-12-30T13:00:00+01:00'
  end: '2022-12-30T14:00:00+01:00'
  value: 1.715
- start: '2022-12-30T14:00:00+01:00'
  end: '2022-12-30T15:00:00+01:00'
  value: 1.732
- start: '2022-12-30T15:00:00+01:00'
  end: '2022-12-30T16:00:00+01:00'
  value: 1.755
- start: '2022-12-30T16:00:00+01:00'
  end: '2022-12-30T17:00:00+01:00'
  value: 1.762
- start: '2022-12-30T17:00:00+01:00'
  end: '2022-12-30T18:00:00+01:00'
  value: 2.355
- start: '2022-12-30T18:00:00+01:00'
  end: '2022-12-30T19:00:00+01:00'
  value: 2.329
- start: '2022-12-30T19:00:00+01:00'
  end: '2022-12-30T20:00:00+01:00'
  value: 2.301
- start: '2022-12-30T20:00:00+01:00'
  end: '2022-12-30T21:00:00+01:00'
  value: 1.673
- start: '2022-12-30T21:00:00+01:00'
  end: '2022-12-30T22:00:00+01:00'
  value: 1.502
- start: '2022-12-30T22:00:00+01:00'
  end: '2022-12-30T23:00:00+01:00'
  value: 1.494
- start: '2022-12-30T23:00:00+01:00'
  end: '2022-12-31T00:00:00+01:00'
  value: 1.488
- start: '2022-12-31T00:00:00+01:00'
  end: '2022-12-31T01:00:00+01:00'
  value: 1.507
- start: '2022-12-31T01:00:00+01:00'
  end: '2022-12-31T02:00:00+01:00'
  value: 1.489
- start: '2022-12-31T02:00:00+01:00'
  end: '2022-12-31T03:00:00+01:00'
  value: 1.488
- start: '2022-12-31T03:00:00+01:00'
  end: '2022-12-31T04:00:00+01:00'
  value: 1.488
- start: '2022-12-31T04:00:00+01:00'
  end: '2022-12-31T05:00:00+01:00'
  value: 1.488
- start: '2022-12-31T05:00:00+01:00'
  end: '2022-12-31T06:00:00+01:00'
  value: 1.488
- start: '2022-12-31T06:00:00+01:00'
  end: '2022-12-31T07:00:00+01:00'
  value: 1.488
- start: '2022-12-31T07:00:00+01:00'
  end: '2022-12-31T08:00:00+01:00'
  value: 1.488
- start: '2022-12-31T08:00:00+01:00'
  end: '2022-12-31T09:00:00+01:00'
  value: 1.494
- start: '2022-12-31T09:00:00+01:00'
  end: '2022-12-31T10:00:00+01:00'
  value: 1.507
- start: '2022-12-31T10:00:00+01:00'
  end: '2022-12-31T11:00:00+01:00'
  value: 1.519
- start: '2022-12-31T11:00:00+01:00'
  end: '2022-12-31T12:00:00+01:00'
  value: 1.526
- start: '2022-12-31T12:00:00+01:00'
  end: '2022-12-31T13:00:00+01:00'
  value: 1.582
- start: '2022-12-31T13:00:00+01:00'
  end: '2022-12-31T14:00:00+01:00'
  value: 1.587
- start: '2022-12-31T14:00:00+01:00'
  end: '2022-12-31T15:00:00+01:00'
  value: 1.635
- start: '2022-12-31T15:00:00+01:00'
  end: '2022-12-31T16:00:00+01:00'
  value: 1.732
- start: '2022-12-31T16:00:00+01:00'
  end: '2022-12-31T17:00:00+01:00'
  value: 1.736
- start: '2022-12-31T17:00:00+01:00'
  end: '2022-12-31T18:00:00+01:00'
  value: 2.307
- start: '2022-12-31T18:00:00+01:00'
  end: '2022-12-31T19:00:00+01:00'
  value: 2.313
- start: '2022-12-31T19:00:00+01:00'
  end: '2022-12-31T20:00:00+01:00'
  value: 2.209
- start: '2022-12-31T20:00:00+01:00'
  end: '2022-12-31T21:00:00+01:00'
  value: 1.596
- start: '2022-12-31T21:00:00+01:00'
  end: '2022-12-31T22:00:00+01:00'
  value: 1.627
- start: '2022-12-31T22:00:00+01:00'
  end: '2022-12-31T23:00:00+01:00'
  value: 1.581
- start: '2022-12-31T23:00:00+01:00'
  end: '2023-01-01T00:00:00+01:00'
  value: 1.533

charging_schedule: 
- start: '2022-12-30T00:00:00+01:00'
  end: '2022-12-30T01:00:00+01:00'
  value: 0
- start: '2022-12-30T01:00:00+01:00'
  end: '2022-12-30T02:00:00+01:00'
  value: 0
- start: '2022-12-30T02:00:00+01:00'
  end: '2022-12-30T03:00:00+01:00'
  value: 0
- start: '2022-12-30T03:00:00+01:00'
  end: '2022-12-30T04:00:00+01:00'
  value: 0
- start: '2022-12-30T04:00:00+01:00'
  end: '2022-12-30T05:00:00+01:00'
  value: 0
- start: '2022-12-30T05:00:00+01:00'
  end: '2022-12-30T06:00:00+01:00'
  value: 0
- start: '2022-12-30T06:00:00+01:00'
  end: '2022-12-30T07:00:00+01:00'
  value: 0
- start: '2022-12-30T07:00:00+01:00'
  end: '2022-12-30T08:00:00+01:00'
  value: 0
- start: '2022-12-30T08:00:00+01:00'
  end: '2022-12-30T09:00:00+01:00'
  value: 0
- start: '2022-12-30T09:00:00+01:00'
  end: '2022-12-30T10:00:00+01:00'
  value: 0
- start: '2022-12-30T10:00:00+01:00'
  end: '2022-12-30T11:00:00+01:00'
  value: 0
- start: '2022-12-30T11:00:00+01:00'
  end: '2022-12-30T12:00:00+01:00'
  value: 0
- start: '2022-12-30T12:00:00+01:00'
  end: '2022-12-30T13:00:00+01:00'
  value: 0
- start: '2022-12-30T13:00:00+01:00'
  end: '2022-12-30T14:00:00+01:00'
  value: 0
- start: '2022-12-30T14:00:00+01:00'
  end: '2022-12-30T15:00:00+01:00'
  value: 0
- start: '2022-12-30T15:00:00+01:00'
  end: '2022-12-30T16:00:00+01:00'
  value: 0
- start: '2022-12-30T16:00:00+01:00'
  end: '2022-12-30T17:00:00+01:00'
  value: 0
- start: '2022-12-30T17:00:00+01:00'
  end: '2022-12-30T18:00:00+01:00'
  value: 0
- start: '2022-12-30T18:00:00+01:00'
  end: '2022-12-30T19:00:00+01:00'
  value: 0
- start: '2022-12-30T19:00:00+01:00'
  end: '2022-12-30T20:00:00+01:00'
  value: 0
- start: '2022-12-30T20:00:00+01:00'
  end: '2022-12-30T21:00:00+01:00'
  value: 0
- start: '2022-12-30T21:00:00+01:00'
  end: '2022-12-30T22:00:00+01:00'
  value: 0
- start: '2022-12-30T22:00:00+01:00'
  end: '2022-12-30T23:00:00+01:00'
  value: 0
- start: '2022-12-30T23:00:00+01:00'
  end: '2022-12-31T00:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T00:00:00+01:00'
  end: '2022-12-31T01:00:00+01:00'
  value: 0
- start: '2022-12-31T01:00:00+01:00'
  end: '2022-12-31T02:00:00+01:00'
  value: 0
- start: '2022-12-31T02:00:00+01:00'
  end: '2022-12-31T03:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T03:00:00+01:00'
  end: '2022-12-31T04:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T04:00:00+01:00'
  end: '2022-12-31T05:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T05:00:00+01:00'
  end: '2022-12-31T06:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T06:00:00+01:00'
  end: '2022-12-31T07:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T07:00:00+01:00'
  end: '2022-12-31T08:00:00+01:00'
  value: 0
- start: '2022-12-31T08:00:00+01:00'
  end: '2022-12-31T09:00:00+01:00'
  value: 0
- start: '2022-12-31T09:00:00+01:00'
  end: '2022-12-31T10:00:00+01:00'
  value: 0
- start: '2022-12-31T10:00:00+01:00'
  end: '2022-12-31T11:00:00+01:00'
  value: 0
- start: '2022-12-31T11:00:00+01:00'
  end: '2022-12-31T12:00:00+01:00'
  value: 0
- start: '2022-12-31T12:00:00+01:00'
  end: '2022-12-31T13:00:00+01:00'
  value: 0
- start: '2022-12-31T13:00:00+01:00'
  end: '2022-12-31T14:00:00+01:00'
  value: 0
- start: '2022-12-31T14:00:00+01:00'
  end: '2022-12-31T15:00:00+01:00'
  value: 0
- start: '2022-12-31T15:00:00+01:00'
  end: '2022-12-31T16:00:00+01:00'
  value: 0
- start: '2022-12-31T16:00:00+01:00'
  end: '2022-12-31T17:00:00+01:00'
  value: 0
- start: '2022-12-31T17:00:00+01:00'
  end: '2022-12-31T18:00:00+01:00'
  value: 0
- start: '2022-12-31T18:00:00+01:00'
  end: '2022-12-31T19:00:00+01:00'
  value: 0
- start: '2022-12-31T19:00:00+01:00'
  end: '2022-12-31T20:00:00+01:00'
  value: 0
- start: '2022-12-31T20:00:00+01:00'
  end: '2022-12-31T21:00:00+01:00'
  value: 0
- start: '2022-12-31T21:00:00+01:00'
  end: '2022-12-31T22:00:00+01:00'
  value: 0
- start: '2022-12-31T22:00:00+01:00'
  end: '2022-12-31T23:00:00+01:00'
  value: 0
- start: '2022-12-31T23:00:00+01:00'
  end: '2023-01-01T00:00:00+01:00'
  value: 0

icon: mdi:flash
friendly_name: EV Smart Charging Charging
```

Attributes of the sensor with continuous enabled
```
current_price: 2.329
EV SOC: 92
EV target SOC: 100.0
Charging is planned: true
Charging start time: 2022-12-30T23:00:00+01:00
Charging stop time: 2022-12-31T00:00:00+01:00
Charging number of hours: 1
raw_two_days: 
- start: '2022-12-30T00:00:00+01:00'
  end: '2022-12-30T01:00:00+01:00'
  value: 1.584
- start: '2022-12-30T01:00:00+01:00'
  end: '2022-12-30T02:00:00+01:00'
  value: 1.536
- start: '2022-12-30T02:00:00+01:00'
  end: '2022-12-30T03:00:00+01:00'
  value: 1.517
- start: '2022-12-30T03:00:00+01:00'
  end: '2022-12-30T04:00:00+01:00'
  value: 1.494
- start: '2022-12-30T04:00:00+01:00'
  end: '2022-12-30T05:00:00+01:00'
  value: 1.494
- start: '2022-12-30T05:00:00+01:00'
  end: '2022-12-30T06:00:00+01:00'
  value: 1.532
- start: '2022-12-30T06:00:00+01:00'
  end: '2022-12-30T07:00:00+01:00'
  value: 1.591
- start: '2022-12-30T07:00:00+01:00'
  end: '2022-12-30T08:00:00+01:00'
  value: 1.713
- start: '2022-12-30T08:00:00+01:00'
  end: '2022-12-30T09:00:00+01:00'
  value: 1.746
- start: '2022-12-30T09:00:00+01:00'
  end: '2022-12-30T10:00:00+01:00'
  value: 1.759
- start: '2022-12-30T10:00:00+01:00'
  end: '2022-12-30T11:00:00+01:00'
  value: 1.731
- start: '2022-12-30T11:00:00+01:00'
  end: '2022-12-30T12:00:00+01:00'
  value: 1.718
- start: '2022-12-30T12:00:00+01:00'
  end: '2022-12-30T13:00:00+01:00'
  value: 1.713
- start: '2022-12-30T13:00:00+01:00'
  end: '2022-12-30T14:00:00+01:00'
  value: 1.715
- start: '2022-12-30T14:00:00+01:00'
  end: '2022-12-30T15:00:00+01:00'
  value: 1.732
- start: '2022-12-30T15:00:00+01:00'
  end: '2022-12-30T16:00:00+01:00'
  value: 1.755
- start: '2022-12-30T16:00:00+01:00'
  end: '2022-12-30T17:00:00+01:00'
  value: 1.762
- start: '2022-12-30T17:00:00+01:00'
  end: '2022-12-30T18:00:00+01:00'
  value: 2.355
- start: '2022-12-30T18:00:00+01:00'
  end: '2022-12-30T19:00:00+01:00'
  value: 2.329
- start: '2022-12-30T19:00:00+01:00'
  end: '2022-12-30T20:00:00+01:00'
  value: 2.301
- start: '2022-12-30T20:00:00+01:00'
  end: '2022-12-30T21:00:00+01:00'
  value: 1.673
- start: '2022-12-30T21:00:00+01:00'
  end: '2022-12-30T22:00:00+01:00'
  value: 1.502
- start: '2022-12-30T22:00:00+01:00'
  end: '2022-12-30T23:00:00+01:00'
  value: 1.494
- start: '2022-12-30T23:00:00+01:00'
  end: '2022-12-31T00:00:00+01:00'
  value: 1.488
- start: '2022-12-31T00:00:00+01:00'
  end: '2022-12-31T01:00:00+01:00'
  value: 1.507
- start: '2022-12-31T01:00:00+01:00'
  end: '2022-12-31T02:00:00+01:00'
  value: 1.489
- start: '2022-12-31T02:00:00+01:00'
  end: '2022-12-31T03:00:00+01:00'
  value: 1.488
- start: '2022-12-31T03:00:00+01:00'
  end: '2022-12-31T04:00:00+01:00'
  value: 1.488
- start: '2022-12-31T04:00:00+01:00'
  end: '2022-12-31T05:00:00+01:00'
  value: 1.488
- start: '2022-12-31T05:00:00+01:00'
  end: '2022-12-31T06:00:00+01:00'
  value: 1.488
- start: '2022-12-31T06:00:00+01:00'
  end: '2022-12-31T07:00:00+01:00'
  value: 1.488
- start: '2022-12-31T07:00:00+01:00'
  end: '2022-12-31T08:00:00+01:00'
  value: 1.488
- start: '2022-12-31T08:00:00+01:00'
  end: '2022-12-31T09:00:00+01:00'
  value: 1.494
- start: '2022-12-31T09:00:00+01:00'
  end: '2022-12-31T10:00:00+01:00'
  value: 1.507
- start: '2022-12-31T10:00:00+01:00'
  end: '2022-12-31T11:00:00+01:00'
  value: 1.519
- start: '2022-12-31T11:00:00+01:00'
  end: '2022-12-31T12:00:00+01:00'
  value: 1.526
- start: '2022-12-31T12:00:00+01:00'
  end: '2022-12-31T13:00:00+01:00'
  value: 1.582
- start: '2022-12-31T13:00:00+01:00'
  end: '2022-12-31T14:00:00+01:00'
  value: 1.587
- start: '2022-12-31T14:00:00+01:00'
  end: '2022-12-31T15:00:00+01:00'
  value: 1.635
- start: '2022-12-31T15:00:00+01:00'
  end: '2022-12-31T16:00:00+01:00'
  value: 1.732
- start: '2022-12-31T16:00:00+01:00'
  end: '2022-12-31T17:00:00+01:00'
  value: 1.736
- start: '2022-12-31T17:00:00+01:00'
  end: '2022-12-31T18:00:00+01:00'
  value: 2.307
- start: '2022-12-31T18:00:00+01:00'
  end: '2022-12-31T19:00:00+01:00'
  value: 2.313
- start: '2022-12-31T19:00:00+01:00'
  end: '2022-12-31T20:00:00+01:00'
  value: 2.209
- start: '2022-12-31T20:00:00+01:00'
  end: '2022-12-31T21:00:00+01:00'
  value: 1.596
- start: '2022-12-31T21:00:00+01:00'
  end: '2022-12-31T22:00:00+01:00'
  value: 1.627
- start: '2022-12-31T22:00:00+01:00'
  end: '2022-12-31T23:00:00+01:00'
  value: 1.581
- start: '2022-12-31T23:00:00+01:00'
  end: '2023-01-01T00:00:00+01:00'
  value: 1.533

charging_schedule: 
- start: '2022-12-30T00:00:00+01:00'
  end: '2022-12-30T01:00:00+01:00'
  value: 0
- start: '2022-12-30T01:00:00+01:00'
  end: '2022-12-30T02:00:00+01:00'
  value: 0
- start: '2022-12-30T02:00:00+01:00'
  end: '2022-12-30T03:00:00+01:00'
  value: 0
- start: '2022-12-30T03:00:00+01:00'
  end: '2022-12-30T04:00:00+01:00'
  value: 0
- start: '2022-12-30T04:00:00+01:00'
  end: '2022-12-30T05:00:00+01:00'
  value: 0
- start: '2022-12-30T05:00:00+01:00'
  end: '2022-12-30T06:00:00+01:00'
  value: 0
- start: '2022-12-30T06:00:00+01:00'
  end: '2022-12-30T07:00:00+01:00'
  value: 0
- start: '2022-12-30T07:00:00+01:00'
  end: '2022-12-30T08:00:00+01:00'
  value: 0
- start: '2022-12-30T08:00:00+01:00'
  end: '2022-12-30T09:00:00+01:00'
  value: 0
- start: '2022-12-30T09:00:00+01:00'
  end: '2022-12-30T10:00:00+01:00'
  value: 0
- start: '2022-12-30T10:00:00+01:00'
  end: '2022-12-30T11:00:00+01:00'
  value: 0
- start: '2022-12-30T11:00:00+01:00'
  end: '2022-12-30T12:00:00+01:00'
  value: 0
- start: '2022-12-30T12:00:00+01:00'
  end: '2022-12-30T13:00:00+01:00'
  value: 0
- start: '2022-12-30T13:00:00+01:00'
  end: '2022-12-30T14:00:00+01:00'
  value: 0
- start: '2022-12-30T14:00:00+01:00'
  end: '2022-12-30T15:00:00+01:00'
  value: 0
- start: '2022-12-30T15:00:00+01:00'
  end: '2022-12-30T16:00:00+01:00'
  value: 0
- start: '2022-12-30T16:00:00+01:00'
  end: '2022-12-30T17:00:00+01:00'
  value: 0
- start: '2022-12-30T17:00:00+01:00'
  end: '2022-12-30T18:00:00+01:00'
  value: 0
- start: '2022-12-30T18:00:00+01:00'
  end: '2022-12-30T19:00:00+01:00'
  value: 0
- start: '2022-12-30T19:00:00+01:00'
  end: '2022-12-30T20:00:00+01:00'
  value: 0
- start: '2022-12-30T20:00:00+01:00'
  end: '2022-12-30T21:00:00+01:00'
  value: 0
- start: '2022-12-30T21:00:00+01:00'
  end: '2022-12-30T22:00:00+01:00'
  value: 0
- start: '2022-12-30T22:00:00+01:00'
  end: '2022-12-30T23:00:00+01:00'
  value: 0
- start: '2022-12-30T23:00:00+01:00'
  end: '2022-12-31T00:00:00+01:00'
  value: 1.7662499999999999
- start: '2022-12-31T00:00:00+01:00'
  end: '2022-12-31T01:00:00+01:00'
  value: 0
- start: '2022-12-31T01:00:00+01:00'
  end: '2022-12-31T02:00:00+01:00'
  value: 0
- start: '2022-12-31T02:00:00+01:00'
  end: '2022-12-31T03:00:00+01:00'
  value: 0
- start: '2022-12-31T03:00:00+01:00'
  end: '2022-12-31T04:00:00+01:00'
  value: 0
- start: '2022-12-31T04:00:00+01:00'
  end: '2022-12-31T05:00:00+01:00'
  value: 0
- start: '2022-12-31T05:00:00+01:00'
  end: '2022-12-31T06:00:00+01:00'
  value: 0
- start: '2022-12-31T06:00:00+01:00'
  end: '2022-12-31T07:00:00+01:00'
  value: 0
- start: '2022-12-31T07:00:00+01:00'
  end: '2022-12-31T08:00:00+01:00'
  value: 0
- start: '2022-12-31T08:00:00+01:00'
  end: '2022-12-31T09:00:00+01:00'
  value: 0
- start: '2022-12-31T09:00:00+01:00'
  end: '2022-12-31T10:00:00+01:00'
  value: 0
- start: '2022-12-31T10:00:00+01:00'
  end: '2022-12-31T11:00:00+01:00'
  value: 0
- start: '2022-12-31T11:00:00+01:00'
  end: '2022-12-31T12:00:00+01:00'
  value: 0
- start: '2022-12-31T12:00:00+01:00'
  end: '2022-12-31T13:00:00+01:00'
  value: 0
- start: '2022-12-31T13:00:00+01:00'
  end: '2022-12-31T14:00:00+01:00'
  value: 0
- start: '2022-12-31T14:00:00+01:00'
  end: '2022-12-31T15:00:00+01:00'
  value: 0
- start: '2022-12-31T15:00:00+01:00'
  end: '2022-12-31T16:00:00+01:00'
  value: 0
- start: '2022-12-31T16:00:00+01:00'
  end: '2022-12-31T17:00:00+01:00'
  value: 0
- start: '2022-12-31T17:00:00+01:00'
  end: '2022-12-31T18:00:00+01:00'
  value: 0
- start: '2022-12-31T18:00:00+01:00'
  end: '2022-12-31T19:00:00+01:00'
  value: 0
- start: '2022-12-31T19:00:00+01:00'
  end: '2022-12-31T20:00:00+01:00'
  value: 0
- start: '2022-12-31T20:00:00+01:00'
  end: '2022-12-31T21:00:00+01:00'
  value: 0
- start: '2022-12-31T21:00:00+01:00'
  end: '2022-12-31T22:00:00+01:00'
  value: 0
- start: '2022-12-31T22:00:00+01:00'
  end: '2022-12-31T23:00:00+01:00'
  value: 0
- start: '2022-12-31T23:00:00+01:00'
  end: '2023-01-01T00:00:00+01:00'
  value: 0

icon: mdi:flash
friendly_name: EV Smart Charging Charging
```

**Comment by jonasbkarlsson** [comment]
Thanks for reporting. It's related to non-continous hours and many hours with exactly the same price. It's not related to crossing midnight.

**Comment by jonasbkarlsson** [comment]
Fixed in main branch #85. Will be included in next release.
